Gjoranv/new metrics proxy fixes

gjoranv opened the pull request.

gjoranv committed:
Fix and simplify generation of absolute jar file path.

+ add missing unit test

gjoranv committed:
Add module to CMakeLists

gjoranv committed:
Remove one duplicate of monitoring config default values.

- They are still duplicated in DomAdminBuilderBase.
- Align def file value with default in DomAdminBuilderBase.

gjoranv committed:
Package container-di as a bundle

- Necessary for bundle-plugin to generate Import-Package for its
  packages when used as a provided dependency.

gjoranv committed:
Export package com.yahoo.stream

gjoranv committed:
Add config definitions to CMakeLists

- They must be pre-installed, because we only pick up config defs
  from application package bundles, and not bundles installed
  from a container cluster.

bjorncs commented on config-model/src/main/java/com/yahoo/vespa/model/admin/metricsproxy/MetricsProxyContainerCluster.java:
```diff
@@ -150,19 +147,16 @@ private MetricSet getAdditionalDefaultMetrics() {
     }
 
-    private int getIntervalMinutes() {
+    private Optional<Integer> getIntervalMinutes() {
```
Consider using `OptionalInt`

bjorncs reviewed (approved):
👍

gjoranv committed:
Allow built-in config definitions without 'com.yahoo' prefix.

- The hardcoded package prefix prevented lookup of producers
  for configs without the prefix.
+ Fix and refactor to use the already resolved builder class
  instead Config(Definition)Key.

gjoranv committed:
Use compile scope for org.json.

- Not an OSGi bundle, and only exported from container-disc.

gjoranv committed:
Add provided dep for jdisc_core to get Import-Packages.

bjorncs commented on metrics-proxy/pom.xml:
```diff
@@ -64,6 +64,13 @@
       <scope>provided</scope>
     </dependency>
+    <dependency>
+      <!-- Not directly used in this module, but needed to get Import-Packages for JDK packages it exports. -->
```
Is jdisc_core always required as provided dependency for bundles?

gjoranv replied:
It exports 'javax.*' packages and so on, so I would in general say 'yes'.

bjorncs replied:
OK, so it is not necessary for java.* packages.

gjoranv replied:
No, because OSGi (at least in r7) still allows not importing them.
